dingo/pyoptInterface_based_impl.py

```
@@ -0,0 +1,499 @@
+    except Exception as e:
+        print(f"An unexpected error occurred: {e}")
+
+
+def inner_ball(A, b, solver_name=None):
+    """A Python function to compute the maximum inscribed ball in the given polytope using PyOptInterface LP modeler
+    Returns the optimal solution for the following linear program:
+    max r, subject to,
+    a_ix + r||a_i|| <= b, i=1,...,n
+
+    Keyword arguments:
+    A -- an mxn matrix that contains the normal vectors of the facets of the polytope row-wise
+    b -- a m-dimensional vector
+    solver_name  -- the solver to use, i.e., a string of the solver name (if None, the default solver is used)
+    """
+
+    extra_column = []
+
+    m = A.shape[0]
+    n = A.shape[1]
+
+    for i in range(A.shape[0]):
+        entry = np.linalg.norm(A[i])
+        extra_column.append(entry)
+
+    column = np.asarray(extra_column)
+    A_expand = np.c_[A, column]
+
+    if solver_name is None:
+        solver_name = default_solver
+    SOLVER = get_solver(solver_name)
+    model = SOLVER.Model()
+    model.set_model_attribute(poi.ModelAttribute.Silent, True)
+
+    # Create variables where x[n] is the radius
+    x = np.empty(n + 1, dtype=object)
+    for i in range(n + 1):
+        x[i] = model.add_variable()
+
+    # Add the constraints a_ix + r||a_i|| <= b
+    for i in range(m):
+        model.add_linear_constraint(dot(A_expand[i], x), poi.Leq, b[i])
+
+    # Set the objective function
+    obj = poi.ExprBuilder(x[n])
+    model.set_objective(obj, sense=poi.ObjectiveSense.Maximize)
+
+    # Optimize model
+    model.optimize()
+
+    # Get the center point and the radius of max ball from the solution of LP
+    point = [model.get_value(x[i]) for i in range(n)]
+
+    # Get radius
+    r = model.get_value(obj)
+
+    # And check whether the computed radius is negative
+    if r < 0:
+        print(
+            "The radius calculated has negative value. The polytope is infeasible or something went wrong with the solver"
```

Could you please throw an exception here instead of printing a message?

---

OK.